packages/loader/container-loader/src/container.ts

```diff
@@ -626,6 +627,11 @@ export class Container
 		return this._deltaManager.readOnlyInfo;
 	}
 
+	public get containerMetadata(): Record<string, unknown> | undefined {
+		// Provide a copy instead of the actual object.
+		return { ...this._containerMetadata };
```

Comment:
I like the copy part :) I think we can also use Readonly<Record<...>> to force a copy at consumer side if they need to modify it.
Q: are records truly unknown, or just strings? Even if we want to support more complex types in the future, but all cases are strings today, I'd probably just use string type. It adds a bit more confidence that shallow copy (as opposed to deep copy) is sufficient here.

Reply:
Yes, currently it is just string. Changing it to readonly  and removing copying from here.